### PR TITLE
Additional fix mgmt, support convenience API for publicNetworkAccess

### PR DESCRIPTION
**Pull request by v-hongli1**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-sql/src/main/java/com/azure/resourcemanager/sql/models/SqlServer.java [hunk 1]
```diff
@@ -75,6 +75,13 @@ public interface SqlServer
      */
     SqlFirewallRule enableAccessFromAzureServices();
 
+    /**
+     * Whether the sql server can be accessed from public network.
+     *
+     * @return whether the disk can be accessed from sql server.
+     */
+    ServerNetworkAccessFlag publicNetworkAccess();
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-sql/src/main/java/com/azure/resourcemanager/sql/models/SqlServer.java:83]: OK.. sql uses a strange name.

Fine for SDK.

File: sdk/resourcemanager/azure-resourcemanager-sql/src/main/java/com/azure/resourcemanager/sql/models/SqlServer.java [hunk 2]
```diff
@@ -268,6 +276,17 @@ SqlVirtualNetworkRule.DefinitionStages.Blank<? extends WithCreate> defineVirtual
                 String virtualNetworkRuleName);
         }
 
+
+        /** The stage of sql server definition allowing to configure network access settings. */
+        interface WithPublicNetworkAccess {
+            /**
+             * Disables public network access for the sql server.
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-sql/src/main/java/com/azure/resourcemanager/sql/models/SqlServer.java:283]: SQL Server

File: sdk/resourcemanager/azure-resourcemanager-sql/src/main/java/com/azure/resourcemanager/sql/models/SqlServer.java [hunk 3]
```diff
@@ -75,6 +75,13 @@ public interface SqlServer
      */
     SqlFirewallRule enableAccessFromAzureServices();
 
+    /**
+     * Whether the sql server can be accessed from public network.
+     *
+     * @return whether the disk can be accessed from sql server.
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-sql/src/main/java/com/azure/resourcemanager/sql/models/SqlServer.java:81]: SQL Server

**Review by weidongxu-microsoft — COMMENTED** [review]
CHANGELOG as well.

File: sdk/resourcemanager/azure-resourcemanager-sql/src/main/java/com/azure/resourcemanager/sql/models/SqlServer.java (hunk 3 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-sql/src/main/java/com/azure/resourcemanager/sql/models/SqlServer.java]: Fixed in the new version.

File: sdk/resourcemanager/azure-resourcemanager-appservice/CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,9 @@
 ### Features Added
 
 - Supported disabling public network access in `WebApp` via `disablePublicNetworkAccess()`, for private link feature.
+- Supported disabling public network access in `FunctionApp` via `disablePublicNetworkAccess()`, for private link feature.
+- Supported disabling public network access in `DeploymentSlot` via `disablePublicNetworkAccess()`, for private link feature.
+- Supported disabling public network access in `FunctionDeploymentSlot` via `disablePublicNetworkAccess()`, for private link feature.
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-appservice/CHANGELOG.md:10]: Seems we added support for snaptshot, search and sql as well. We should reflect them in the changelog.

Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-appservice/CHANGELOG.md:10]: Never mind... I saw them in their changelogs.

**Review by XiaofeiCao — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
need refresh and move changelog items to 2.29.0-beta.1

**Pull request closed without merging**